zebrad/src/components/tokio.rs

```diff
@@ -46,16 +46,22 @@ pub(crate) trait RuntimeRun {
 impl RuntimeRun for Runtime {
     fn run(&mut self, fut: impl Future<Output = Result<(), Report>>) {
         let result = self.block_on(async move {
-            // If the run task and shutdown are both ready, select! chooses
-            // one of them at random.
+            // Always poll the shutdown future first.
+            //
+            // Otherwise, a busy Zebra instance could starve the shutdown future,
+            // and delay shutting down.
             tokio::select! {
-                result = fut => result,
+                biased;
```

Comment:
Cool, I didn't know about that

Reply:
`select! { biased; ... }` was added our big tokio update, it wasn't available before:
https://docs.rs/tokio/1.14.0/tokio/macro.select.html#fairness